presto-prometheus/src/main/java/com/facebook/presto/plugin/prometheus/PrometheusQueryResponse.java

```diff
@@ -0,0 +1,132 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.plugin.prometheus;
+
+import com.facebook.presto.spi.PrestoException;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.facebook.presto.plugin.prometheus.PrometheusErrorCode.PROMETHEUS_PARSE_ERROR;
+import static java.util.Collections.singletonList;
+
+public class PrometheusQueryResponse
+{
+    private boolean status;
+
+    private String error;
+    private String errorType;
+    private String resultType;
+    private String result;
+    private List<PrometheusMetricResult> results;
+
+    public PrometheusQueryResponse(InputStream response)
+            throws IOException
+    {
+        parsePrometheusQueryResponse(response);
+    }
+
+    private void parsePrometheusQueryResponse(InputStream response)
+            throws IOException
+    {
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.registerModule(new JavaTimeModule());
+        JsonParser parser = new JsonFactory().createParser(response);
+        while (!parser.isClosed()) {
+            JsonToken jsonToken = parser.nextToken();
+            if (JsonToken.FIELD_NAME.equals(jsonToken)) {
+                if (parser.getCurrentName().equals("status")) {
+                    parser.nextToken();
+                    if (parser.getValueAsString().equals("success")) {
+                        this.status = true;
+                        while (!parser.isClosed()) {
+                            parser.nextToken();
+                            if (JsonToken.FIELD_NAME.equals(jsonToken)) {
+                                if (parser.getCurrentName().equals("resultType")) {
+                                    parser.nextToken();
+                                    resultType = parser.getValueAsString();
+                                }
+                                if (parser.getCurrentName().equals("result")) {
+                                    parser.nextToken();
+                                    ArrayNode node = mapper.readTree(parser);
+                                    result = node.toString();
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                    else {
+                        //error path
+                        String parsedStatus = parser.getValueAsString();
+                        parser.nextToken();
```

Comment:
why calling 2 consecutive nextToken()? add comments?

Reply:
parsing json is key-value based, so first nextToken is advanced to the key, nextToken advances to the value. Added comment for clarification